### PR TITLE
Added tests for azblob/service

### PR DESCRIPTION
**Pull request by gapra-msft**
Adding tests for service client.
Added infra to allow connection string clients to be recorded

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to [CHANGELOG.md][] are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/storage/azblob/testdata/recordings/Test/TestSASContainerClient.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "Entries": [],
```
Comment (gapra-msft) [inline sdk/storage/azblob/testdata/recordings/Test/TestSASContainerClient.json:2]: TODO: Why is this empty even though it calls REST API?

Comment (tasherif-msft) [reply sdk/storage/azblob/testdata/recordings/Test/TestSASContainerClient.json:2]: my guess: sas parameter not being parsed properly somehow

**Comment by jhendrixMSFT** [comment]
You'll want to rebase on top of `main` so that you can use test assets and remove the checked in recordings.

**Pull request closed without merging**